common/services/user-locations.js

```diff
@@ -1,53 +1,47 @@
 const axios = require('axios')
-const { map } = require('lodash')
 
 const { AUTH_PROVIDERS, NOMIS_ELITE2_API } = require('../../config')
 const { decodeAccessToken } = require('../lib/access-token')
-const referenceDataService = require('./reference-data')
+const { getLocationsByNomisAgencyId } = require('./reference-data')
+const { buildAuthHeader } = require('../lib/request')
 
 function getUserLocations(token) {
-  return _getAgencies(token).then(agencies =>
-    referenceDataService.getLocationsByNomisAgencyId(agencies)
-  )
-}
+  const { auth_source: authSource } = decodeAccessToken(token)
 
-function _getAgencies(token) {
-  const { auth_source: source } = decodeAccessToken(token)
+  if (authSource === 'auth') {
+    return getAuthAgencies(token).then(agencies =>
+      getLocationsByNomisAgencyId(agencies)
+    )
+  }
 
-  switch (source) {
-    case 'auth':
-      return _getAuthAgencies(token)
-    case 'nomis':
-      return _getNomisAgencies(token)
-    default:
-      return Promise.resolve([])
+  if (authSource === 'nomis') {
+    return getNomisAgencies(token).then(agencies =>
+      getLocationsByNomisAgencyId(agencies)
+    )
   }
+
+  return []
```

Comment:
Just noticed this should return the same type like the current method. The other instances return a promise (which ultimately return an array), but this just returns an array. I'd probably keep the `Promise.resolve([])` line to produce a consistent return type.